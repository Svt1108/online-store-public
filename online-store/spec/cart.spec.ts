import ShoppingCart from "../src/components/cart";

describe("Method showCart", () => {
  const testShoppingCart = new ShoppingCart();

  test("should be defined", () => {
    expect(testShoppingCart.showCart).toBeDefined;
  });

  it("renders correctly with empty cart", () => {
    const data = [
      {
        id: "1",
        name: "Lovely Mexico",
        img: "plate-1.jpg",
        type: "stoneware",
        author: "Grazielli David",
        year: "2022",
        color: "chocolate",
        secondColor: "green",
        quantity: "3",
        popularity: "yes",
        price: "10",
      },
      {
        id: "2",
        name: "Coffee time",
        img: "plate-2.jpg",
        type: "stoneware",
        author: "Milena Kibede",
        year: "2021",
        color: "pink",
        secondColor: "red",
        quantity: "5",
        popularity: "no",
        price: "37",
      },
    ];
    testShoppingCart.showCart(data);
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
