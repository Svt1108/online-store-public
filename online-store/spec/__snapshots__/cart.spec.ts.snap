// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Method showCart renders correctly 1`] = `"<div class=\\"modal-wrap\\"><div class=\\"cart\\"><div class=\\"cart__title brown-text text-lighten-2\\">You chose these plates: </div><div class=\\"cart__items\\"></div><div class=\\"cart__total\\"><div>Total:  0 pieces</div><div>0 €</div></div><div class=\\"btn-wrap\\"><button class=\\"btn waves-effect waves-light brown lighten-2\\">Turn to choice</button><button class=\\"btn waves-effect waves-light teal darken-1\\">Buy all</button></div></div></div>"`;

exports[`Method showCart renders correctly with empty cart 1`] = `"<div class=\\"modal-wrap\\"><div class=\\"cart\\"><div class=\\"cart__title brown-text text-lighten-2\\">You chose these plates: </div><div class=\\"cart__items\\"></div><div class=\\"cart__total\\"><div>Total:  0 pieces</div><div>0 €</div></div><div class=\\"btn-wrap\\"><button class=\\"btn waves-effect waves-light brown lighten-2\\">Turn to choice</button><button class=\\"btn waves-effect waves-light teal darken-1\\">Buy all</button></div></div></div>"`;
